src/components/shared/TheExercise.vue

```diff
@@ -22,11 +22,15 @@
     </div>
 </template>
 
-<script>
-export default {
-  name: 'TheExercise',
-  props: ['img', 'name', 'sets', 'reps', 'weight', 'rest']
-}
+<script setup lang="ts">
+defineProps<{
+  name: string,
+  img: string,
+  sets: string,
+  reps: string,
+  weight: string,
+  rest: string
+}>()
```

Comment:
Is it possible to use Exercise type here?

Reply:
Hmm... it doesn't seem like it is, actually 😛

I tried to do that by writing it like this: `defineProps<Exercise>()` importing the Exercise previously, of course, and I got the error (see the attached screen), which I don't understand well but from this error I got here: https://github.com/vuejs/core/issues/4294 and then tried this workaround: https://github.com/vuejs/core/issues/4294#issuecomment-969625519 as well as this: https://github.com/vuejs/core/issues/4294#issuecomment-942838354.

In the latter case I still had the same error, whereas in the first one I got a warning/error about Exercise being a type but not used anywhere.

Apparently, there are also some plugins that for some people solved this issue and for others didn't.

So, finally, I stopped reading here: https://github.com/vuejs/core/issues/4294#issuecomment-1316097560 and decided to ask you what to do 😛

Reply:
I'll take a look into it tomorrow. Check how to use typescript version of defineProps in the meantime.

Reply:
**Checked the docs** and found the info below"
![props](https://user-images.githubusercontent.com/113309294/228048166-3bf2800f-7622-444f-9c56-5779d15534c1.PNG)

So, am I supposed to define the same interface again but in the appropriate component? If so, I could just define this prop like I did already, by passing an object with types, couldn't I?